app/assets/javascripts/modules/MultiTableFilter.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * # Multiple table filter
+ *
+ * Filters across one or more tables, and can also hide the elements associated with a table.
+ *
+ * Requires an element to have a data-dough-component="MultiTableFilter" attribute and an ID.
+ *
+ * Makes use of jquery.fastLiveFilter.
+ * See test fixture for sample markup - /spec/js/fixtures/MultiTableFilter.html
+ */
+
+define(['jquery', 'jqueryFastLiveFilter', 'DoughBaseComponent'],
+       function($, jqueryFastLiveFilter, DoughBaseComponent) {
+  'use strict';
+
+  var MultiTableFilterProto,
+      defaultConfig = {
+        filterTargetSelector: '.js-filterable',
+        filterFieldSelector: '.js-filter-input',
+        filterListSelector: '.js-filter-rows',
+        filterWrapperSelector: '.js-filter-group'
+      };
+
+  function MultiTableFilter($el, config) {
+    MultiTableFilter.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   * @type {[type]}
+   * @private
+   */
+  DoughBaseComponent.extend(MultiTableFilter);
+  MultiTableFilterProto = MultiTableFilter.prototype;
+
+  /**
+   * Init function
+   *
+   * Set up listeners and accept promise
+   *
+   * @param  {[type]} initialised [description]
+   * @return {[type]}             [description]
+   */
+  MultiTableFilterProto.init = function(initialised) {
+    this.bindFilterToElements();
+    this._initialisedSuccess(initialised);
+    return this;
+  };
+
+  /**
+   * show/hide filterWrappers when all filterTargetSelectores are hidden
+   */
```

Comment:
Ah! I've just been guessing the format of these to try to keep them in roughly the same format. Installed the addon so it'll be correct now. May as well go through and make those changes.